lib/active_merchant/billing/gateways/nmi.rb

```diff
@@ -192,8 +192,8 @@ def add_network_token_fields(post, payment_method)
         if payment_method.source == :apple_pay || payment_method.source == :google_pay
           post[:cavv] = payment_method.payment_cryptogram
           post[:eci] = payment_method.eci
-          post[:decrypted_applepay_data] = 1
-          post[:decrypted_googlepay_data] = 1
+          post[:decrypted_applepay_data] = 1 if payment_method.source == :apple_pay
+          post[:decrypted_googlepay_data] = 1 if payment_method.source == :google_pay
```

Comment:
💬 Comment:

just an idea => 

```ruby
post[payment_method.source.to_s.gsub('_','')] = 1
```

Reply:
Looks a little bit complex for what we need, and it is not clear what it is doing. Maybe I prefer a simple and boring way this time.